src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonWebToken.PayloadClaimSet.cs

```diff
@@ -117,5 +128,77 @@ private protected virtual void ReadPayloadValue(ref Utf8JsonReader reader, IDict
                 claims[propertyName] = JsonSerializerPrimitives.ReadPropertyValueAsObject(ref reader, propertyName, JsonClaimSet.ClassName, true);
             }
         }
+
+#if NET8_0_OR_GREATER
+        private protected virtual void ReadPayloadValue(
+            ref Utf8JsonReader reader,
+            Dictionary<string, object> claims,
+            Dictionary<string, (int startIndex, int length)?> claimsBytes,
+            Memory<byte> tokenAsMemory)
+        {
+            if (reader.ValueTextEquals(JwtPayloadUtf8Bytes.Aud))
```

Comment:
Looks like we are assuming that all the reads from the Utf8JsonReader will succeed. If the JSON is malformed or not as expected, this could cause exceptions. Do we need to consider any error handling?

Reply:
The error handling would be as is currently - these errors will just bubble up and be handled in JsonWebToken.